### PR TITLE
Uses datastore to permanently store the comments

### PR DESCRIPTION
**Pull request by tjncgit**
- Implement the Datastore api to replace and overcome the limitations of the ArrayList data structure and provide persistent storage
- Datastore is a noSQL database that offers high performance, auto scaling, and simplifies the application development process

-

### CODE REVIEW

File: portfolio/pom.xml [hunk 1]
```diff
@@ -35,7 +45,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
```
Comment (ohfunc) [inline portfolio/pom.xml:47]: You can remove this TODO.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -6,13 +6,39 @@
...
+                <li>Tokyo</li>
+                <img src='/images/tokyo.jpg' width=500 height=300>>
+            </ul>
+        <h2>Comments</h2>
+        <div id='comments-container'>
+        </div>
+      </div>
+      <div>
+         <form action="/data" method="post">
+         <input type="text" name="input-comment" >
```
Comment (ohfunc) [inline portfolio/src/main/webapp/index.html:38]: Does there need to be a forward slash here?

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -6,13 +6,39 @@
...
+            <p>I was born in Cameroon on April 4th 2000, in the summer of 2005 my family and I moved to America and have been staying in Princes Georges County Maryland ever since.
+                When i was younger I aspired to become a professional basketbll player; this dream was inspired by favorite basketball player of all time, Kobe Bryant. I love basketball but unfortunately
+                It doesn't seem like I will be able to acheive that dream. I am now pursing a career in computer science.
+            </p>
+        <h2>Places I Want to Travel</h2>
+            <ul>
+                <li>Dubai</li>
+                <img src='/images/dubai.jpeg' width=500 height=300>
+                <li>Bora Bora</li>
+                <img src='/images/bora.jpg' width=500 height=300>>
```
Comment (ohfunc) [inline portfolio/src/main/webapp/index.html:28]: Here and below you have two >'s

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -26,3 +26,25 @@ function addRandomGreeting() {
...
+    comments.forEach((comment) => {
+        listComments.appendChild(createListElement(comment));
+        listComments.appendChild(document.createElement('hr'));
+
+    });
+});
+}
+
+function createListElement(text) {
+  const liElement = document.createElement('p');
```
Comment (ohfunc) [inline portfolio/src/main/webapp/script.js:47]: I think I mentioned this elsewhere, but should this be a list tag, or should the function be renamed?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -23,10 +29,51 @@
...
+    Gson gson = new Gson();
+    String json = gson.toJson(commentList);
+    return json;
+  }
+
+    /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (Howon) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:71]: This is great.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -23,10 +29,51 @@
...
+
+@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "input-comment", "");
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("comment", text);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```
Comment (Howon) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:54]: maybe you want to move this to init() and make `datastore` a class member?

**Pull request merged**